hotfix for ignored `--env` flag issue (#269)

sio4 opened the pull request:
Not fully sure if the issue happen again or if the previous patch does not fill all the cases. Anyway, the same issue happens again:

```console
$ soda reset -e test
pop v6.0.2

[POP] 2022/06/04 20:40:03 info - drop app_development (...)
Error: couldn't drop database app_development: error dropping...
```

#### Issue Description
Even though the `--env` (or short `-e`) flag is given, the command just follows the environment variable.

#### Root Cause
It seems like the issue was caused by confusing naming and the behavior of `PersistantPreRun` field of `cobra.Command` and `PersistentFlags()` function. I added a detailed comment on the code so easy to check later.

(One unrelated commit is also included :-)

Related: #282
Fixes #269

stanislas-m commented on soda/cmd/root_integration_test.go:
```diff
@@ -7,27 +7,29 @@ import (
 )
 
-func Test_RootCmd_NoArg(t *testing.T) {
+func Test_RootCmd_Environtment(t *testing.T) {
```
```suggestion
func Test_RootCmd_Environment(t *testing.T) {
```

stanislas-m commented on soda/cmd/root_integration_test.go:
```diff
@@ -7,27 +7,29 @@ import (
-	c.SetArgs([]string{})
+
+	// Fallback on default env
+	c.SetArgs([]string{"help"})
```
`help` is already the default command, do you really need to change this?

stanislas-m commented on soda/cmd/root_integration_test.go:
```diff
@@ -7,27 +7,29 @@ import (
 
 	// Override with GO_ENV
-	c.SetArgs([]string{})
+	c.SetArgs([]string{"help"})
```
`help` is already the default command, do you really need to change this?

sio4 commented:
Hi @stanislas-m! Thanks for the comment.

I added the explanation of the changed test more clearly (I think :-) and added commits to fix

* the typo on the test name, and more test condition
* fix for the Github action (recent windows runner has a memory allocation issue when race detector is on)

Please take a look at the commits and the last comment I added.

Thanks!
